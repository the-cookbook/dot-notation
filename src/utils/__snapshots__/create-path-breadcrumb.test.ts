import createPathBreadcrumb from '../create-path-breadcrumb';

describe('utils/createPathBreadcrumb()', () => {
  it('should match snapshot', () => {
    expect(createPathBreadcrumb).toMatchSnapshot();
  });

  describe('exceptions', () => {
    it('should parse empty "[]" with default array index "[0]"', () => {
      const expectations: Record<string, string[]> = {
        '[][][]': ['[0]', '[0]', '[0]'],
        'person.secret[][][].key': ['person', 'secret', '[0]', '[0]', '[0]', 'key'],
        'person.addresses[].street': ['person', 'addresses', '[0]', 'street'],
        'person.addresses[].number': ['person', 'addresses', '[0]', 'number'],
        'person.addresses[].postalCode': ['person', 'addresses', '[0]', 'postalCode'],
      };

      Object.keys(expectations).forEach((path) => {
        expect(createPathBreadcrumb(path)).toStrictEqual(expectations[path]);
      });
    });
  });

  describe('object', () => {
    const expectations: Record<string, string[]> = {
      person: ['person'],
      'person.name': ['person', 'name'],
      'person.name.firstName': ['person', 'name', 'firstName'],
      'person.address': ['person', 'address'],
    };

    Object.keys(expectations).forEach((path) => {
      it(`should parse "${path} accordingly`, () => {
        expect(createPathBreadcrumb(path)).toStrictEqual(expectations[path]);
      });
    });
  });

  describe('array', () => {
    const expectations: Record<string, string[]> = {
      '[0]': ['[0]'],
      '[0][0]': ['[0]', '[0]'],
      '[1][0][0]': ['[1]', '[0]', '[0]'],
      '[1][1][0][0]': ['[1]', '[1]', '[0]', '[0]'],
    };

    Object.keys(expectations).forEach((path) => {
      it(`should parse "${path} accordingly`, () => {
        expect(createPathBreadcrumb(path)).toStrictEqual(expectations[path]);
      });
    });
  });

  describe('nested', () => {
    const expectations: Record<string, string[]> = {
      person: ['person'],
      'person.name': ['person', 'name'],
      'person.name.firstName': ['person', 'name', 'firstName'],
      'person.addresses[].street': ['person', 'addresses', '[0]', 'street'],
      'person.addresses[].number': ['person', 'addresses', '[0]', 'number'],
      'person.addresses[].postalCode': ['person', 'addresses', '[0]', 'postalCode'],
    };

    Object.keys(expectations).forEach((path) => {
      it(`should parse "${path} accordingly`, () => {
        expect(createPathBreadcrumb(path)).toStrictEqual(expectations[path]);
      });
    });
  });
});
