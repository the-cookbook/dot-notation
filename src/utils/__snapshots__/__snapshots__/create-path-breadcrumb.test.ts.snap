// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/createPathBreadcrumb() should match snapshot 1`] = `[Function]`;
